pkg/resource/replication_group/delta_util.go

```diff
@@ -91,6 +92,14 @@ func engineVersionsMatch(
 		return r.MatchString(latestEV)
 	}
 
+	// if the version is higher than 6, skip the upstream patch version when comparing.
+	// See https://github.com/aws-controllers-k8s/community/issues/1737
+	majorVersion, _ := strconv.Atoi(desiredEV[0:1])
+	if majorVersion >= 6 {
+		r, _ := regexp.Compile(desiredEV + ".*")
+		return r.MatchString(latestEV)
+	}
+
```

Comment:
Instead of using regexes, how about just checking the major version of desired and latest? You can also add a tiny helper function to get the major version number from the string in a stable, safe way:

```go
// versionNumbersFromString takes a version string like "6.2", "6.x" or "7.0.4" and
// returns the major, minor and patch numbers. If no minor or patch numbers are present
// or contain the "x" placeholder, -1 is returned for that version number.
func versionNumbersFromString(version string) (int, int, int) {
    parts := strings.Split(version, ".")
    major := -1
    minor := -1
    patch := -1
    if len(parts) == 0 {
        return major, minor, patch
    }
    major, _ = strconv.Atoi(parts[0])
    if len(parts) > 1 {
        if !strings.EqualFold(parts[1], "x") {
            minor, _ = strconf.Atoi(parts[1])
        }
    }
    if len(parts) > 2 {
        if !strings.EqualFold(parts[2], "x") {
            patch, _ = strconf.Atoi(parts[2])
        }
    }
    return major, minor, patch
}
```

and in the `engineVersionsMatch()` function you can do something like this:

```go
dMaj, dMin, dPatch := versionNumbersFromString(desiredEV)
lMaj, lMin, lPatch := versionNumbersFromString(latestEV)
if dMaj > 5 {
    // if the version is higher than 5, ignore patch version when comparing.
    return dMaj == lMaj && (dMin < 0 || dMin == lMin)
}
```